File: part3/nodejs/helloClient.js
var thrift = require('thrift');
var helloSvc = require('./gen-nodejs/helloSvc.js');

var connection = thrift.createConnection('localhost', 8585, {
   transport: thrift.TBufferedTransport,
   protocol: thrift.TBinaryProtocol
}).on('error', function(error) {
   console.log(error);
}).on("connect", function() {
   var client = thrift.createClient(helloSvc, connection);
   client.getMessage("Thurston Howell", function(error, result) {
      console.log("Msg from server: " + result);
      connection.end();
  });
});

File: part3/cs/helloClient.cs
using System;
using Thrift.Protocol;
using Thrift.Transport;

namespace helloClient
{
    class Program
    {
        static void Main(string[] args)
        {
            TTransport trans = new TSocket("localhost", 8585);
            trans = new TBufferedTransport(trans as TStreamTransport);
            trans.Open();
            TBinaryProtocol proto = new TBinaryProtocol(trans);
            helloSvc.Client client = new helloSvc.Client(proto);
            string result = client.getMessage("Ginger");
            Console.WriteLine("Received from server: " + result);
        }
    }
}
